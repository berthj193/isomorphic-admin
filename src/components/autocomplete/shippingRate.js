import React from 'react';
import PropTypes from 'prop-types';
import withStoresManager from '../apolloProviders/withStoresManager';
import { getAutocompleteList as shippingRateAutocompleteList } from '../../graphql/shippingRates';
import Select, { SelectOption } from '../uielements/select';

export class ShippingRateAutocomplete extends React.Component {
  static propTypes = {
    shipping_rates: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      name: PropTypes.string,
    })),
  };

  static defaultProps = {
    shipping_rates: [],
  };

  dataSource = (this.props.shipping_rates || [])
    .map(({ id, name }) =>
      <SelectOption value={id} key={id}>{`${name}`}</SelectOption>
    );

  render() {
    return (
      <Select
        showSearch
        {...this.props}
      >
        {this.dataSource}
      </Select>
    );
  }
}

export default withStoresManager(ShippingRateAutocomplete,
  [{
    query: shippingRateAutocompleteList,
  }]);
