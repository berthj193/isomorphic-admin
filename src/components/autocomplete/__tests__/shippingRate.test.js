import React from 'react';
import { ShippingRateAutocomplete } from '../shippingRate';
import { shallow } from 'enzyme';

describe('shipping ratess autocomplete component', () => {
  const getComponent = (props = {}) => shallow(<ShippingRateAutocomplete {...props} />);
  const dataObj = {
    id: '1',
    name: 'foo',
  };

  it('datasource should return empty array when there are no elements', () => {
    const instance = getComponent().instance();
    expect(instance.dataSource.length).toBe(0);
  });

  it('datasource should return one element', () => {
    const instance = getComponent({
      shipping_rates: [dataObj],
    }).instance();

    expect(instance.dataSource).toMatchSnapshot();
  });
});
