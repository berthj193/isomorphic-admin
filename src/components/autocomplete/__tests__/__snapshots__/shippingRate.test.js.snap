// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shipping ratess autocomplete component datasource should return one element 1`] = `
Array [
  <Option
    value="1"
>
    foo
</Option>,
]
`;
